.github/workflows/integration-test.yml

```diff
@@ -15,6 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest , windows-latest ]
+        os_bwc_version: [2.4.0, 2.5.0, 2.6.0, 2.7.0, 2.8.0, 2.9.0, 2.10.0, 2.11.0, 2.12.0, 3.0.0]
```

Comment:
Why 3.0.0, but not 2.13.0?

If this introduces too many new CI checks then I think we can verify with min compatible version and previous minor. Is there an issue files around testing with 1.3 or other versions?

Reply:
I think 2.13 may not have been released at the time of adding this test 😅 . I agree with the assessment of keeping it to oldest version and previous minor. I think the earliest windows version released is 2.4, which is why we kept this as the beginning. I think we released 1.3 windows artifacts, so let me add that. Thus, right now the matrix will be [1.3.0, 2.13.0, 3.0.0].

Reply:
nit: wdyt about calling this `datasource_versions`?

Reply:
Renamed. @cwperks I went back and figured out why I couldn't do anything on the 1.3 branch - see this repo I opened on the security backend plugin repo: https://github.com/opensearch-project/security/issues/4281. Since the publish snapshot to maven was not backported, we unfortunately don't have any zips on the 1.3 line. I can do a fast follow up and add it, but then the earliest published would be something like 1.3.16 or even 1.3.17. Keeping it as 2.4.0, 2.13.0 and 3.0.0 for now to maintain the idea of the earliest supported, last minor and current versions.